apps/admin/frontend/src/components/save_file_modal.tsx

```diff
@@ -0,0 +1,261 @@
+import React, { useContext, useState } from 'react';
+import styled from 'styled-components';
+import { basename, join } from 'path';
+import {
+  assert,
+  throwIllegalValue,
+  assertDefined,
+  Result,
+} from '@votingworks/basics';
+import {
+  isElectionManagerAuth,
+  isSystemAdministratorAuth,
+  usbstick,
+} from '@votingworks/utils';
+import type { ExportDataError } from '@votingworks/admin-backend';
+
+import {
+  Button,
+  Modal,
+  H1,
+  P,
+  useExternalStateChangeListener,
+  Font,
+} from '@votingworks/ui';
+
+import { AppContext } from '../contexts/app_context';
+import { Loading } from './loading';
+
+export const UsbImage = styled.img`
+  margin-right: auto;
+  margin-left: auto;
+  height: 200px;
+`;
+
+interface SaveAsButtonProps {
+  onSave: (location: string) => void;
+  options?: KioskBrowser.SaveDialogOptions;
+}
+
+export function SaveAsButton({
+  onSave,
+  options,
+}: SaveAsButtonProps): JSX.Element {
+  async function useSaveDialog() {
+    assert(window.kiosk);
+    const { filePath } = await window.kiosk.showSaveDialog(options);
+    if (filePath) {
+      onSave(filePath);
+    }
+  }
+  return <Button onPress={useSaveDialog}>Save As…</Button>;
+}
+
+export type SaveFileError =
+  | ExportDataError
+  | {
+      type: 'api-error';
+      message: string;
+    };
+
+export type SaveFileResult = Result<string[], SaveFileError>;
+
+export interface SaveFileModalProps {
+  onSave: (location: string) => Promise<SaveFileResult>;
+  onClose: () => void;
+  /**
+   * Title case name for the type of file being saved, e.g. "Batch Results".
+   */
+  fileTypeTitle: string;
+  /**
+   * Lowercase name for the type of file being saved, e.g. "batch results".
+   */
+  fileType: string;
+  /**
+   * Relative to the root of the USB drive.
+   */
+  defaultRelativePath: string;
+}
+
+type SaveFileModalState = 'error' | 'saving' | 'done' | 'init';
+
+export function SaveFileModal({
```

Comment:
This component is largely copied from `save_file_to_usb` but simplified where possible.